server/src/main/java/org/opensearch/search/aggregations/bucket/terms/MultiTermsAggregator.java

```diff
@@ -414,6 +372,74 @@ public void close() {
         }
     }
 
+    static class MultiValuesSourceCollectorImpl implements MultiTermsValuesSourceCollector {
+
+        private final List<InternalValuesSourceCollector> collectors;
+        private final BytesStreamOutput scratch;
+        private final BytesKeyedBucketOrds bucketOrds;
+        private final BucketsAggregator aggregator;
+        private final LeafBucketCollector sub;
+
+        private final boolean collectViaAggregator;
+
+        public MultiValuesSourceCollectorImpl(
+            List<InternalValuesSourceCollector> collectors,
+            BytesStreamOutput scratch,
+            BytesKeyedBucketOrds bucketOrds,
+            BucketsAggregator aggregator,
+            LeafBucketCollector sub
+        ) {
+            this.collectors = collectors;
+            this.scratch = scratch;
+            this.bucketOrds = bucketOrds;
+            this.aggregator = aggregator;
+            this.sub = sub;
+            this.collectViaAggregator = aggregator != null && sub != null;
+        }
+
+        @Override
```

Comment:
Can we please have a javadoc explaining what all are we trying to do here?

Reply:
Done